torch/distributed/_spmd/api.py

```diff
@@ -219,8 +221,117 @@ def f_true():
     finally:
         torch._utils.is_compiling.__code__ = orig_is_compiling_code
 
-
-def compile(module_override: Optional[Dict[Type[Any], Override]] = None):
+@dataclass
+class _CompiledResult:
+    gm: fx.GraphModule
+    mod: nn.Module
+    opt: torch.optim.Optimizer
+    named_states: Dict[str, torch.Tensor]
+    params_and_buffers: List[torch.Tensor]
+
+
+def _compile(
```

Comment:
assuming no code change in this function.

Reply:
Nothing change except that the extra return object `_CompiledResult`.